.github/workflows/haskell.yml

```diff
@@ -18,7 +18,7 @@ on:
         value: ${{ jobs.build.outputs.ATLAS_VERSION }}
 
 permissions:
-  contents: read
+  contents: write
```

Comment:
I would recommend to keep this "read" only.

We should just make the CI build fail in case the formatting is not correct.